certora/specs/ERC20Invariants.spec

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+methods {
+    function totalSupply() external returns uint256 envfree;
+    function balanceOf(address) external returns uint256 envfree;
+    function delegatee(address) external returns address envfree;
+    function delegatedVotingPower(address) external returns uint256 envfree;
+    function upgradeToAndCall(address, bytes) external => NONDET DELETE;
+}
+
+// Paramater for any address that is not the zero address.
+persistent ghost address A {
+    axiom A != 0;
+}
+
+// Partial sum of delegated votes to parameterized address A.
+// sumOfvotes[x] = \sum_{i=0}^{x-1} balances[i] when delegatee[i] == A;
+ghost mapping(mathint => mathint) sumsOfVotes {
+    init_state axiom forall mathint account. sumsOfVotes[account] == 0;
+}
```

Comment:
Can we move `sumOfVotes` to `Delegation.spec` ? It's not really part of ERC20, and we check properties about it in `Delegation.spec`